### ISSUE TITLE
Sum up and label recurring donations during GIVbacks rounds in the GIVbacks API

### ISSUE THREAD

**Issue by laurenluz** [Giveth/giveth-dapps-v2#4758]
For GIVbacks, donations under $4 are excluded from GIVbacks, but in the case of recurring donations... we should check to see if the total donated throughout the GIVbacks round to that project from a recurring donatino is over $4 - so we need to know the sum of the micro donations to that project from that donor via superfluid in a GIVbacks round.

We would also like to have some indication on if a particular donation total was from a recurring donation sum, or from a regular, one time donation.

cc @WhyldWanderer

**Comment by mohammadranjbarz** [comment]
@WhyldWanderer @maryjaf It's on staging and is ready for the test

**Comment by maryjaf** [comment]
I've checked the response of api in different period and with different users, it works as expected
![Image](https://github.com/user-attachments/assets/33d7a77f-0780-48b7-9521-df93df40db4d)


{
    "amount": "4.343614922499999",
    "currency": "GIV",
    "createdAt": "2024-09-26-11:36:19",
    "valueUsd": 4.344499999999998,
    "anonymous": false,
    "bottomRankInRound": 35,
    "givbacksRound": 51100,
    "projectRank": 22,
    "givbackFactor": 0.61470586,
    "valueUsdAfterGivbackFactor": 2.6705896,
    "giverAddress": "0xa1179f64638adb613ddaac32d918eb6beb824104",
    "txHash": "0xa1179f64638adb613ddaac32d918eb6beb824104-0xe6375bc298aeb29d173b2ab359d492439a43b268-0xdfd824f6928b9776c031f7ead948090e2824ce8b-5.0-9.0-1727350578",
    "network": "optimism-sepolia",
    "source": "giveth.io",
    "giverName": "mary jafarimehr",
    "giverEmail": null,
    "projectLink": "https://giveth.io/project/mejan",
    "isReferrerGivbackEligible": false,
    "referrerWallet": null,
    "numberOfStreamedDonations": 234,
    "parentRecurringDonationId": "278",
    "parentRecurringDonationTxHash": "0x49ed78ca2c75faa7a43d621f7ed614c1293df3bf9ea7e147acd41f1b0bc30605"
  },

__________

![Image](https://github.com/user-attachments/assets/a8547983-1062-4181-8d9c-4f4104926d05)

{
    "amount": "2.830136985",
    "currency": "GIV",
    "createdAt": "2024-09-29-07:51:44",
    "valueUsd": 2.8303,
    "anonymous": false,
    "bottomRankInRound": 1,
    "givbacksRound": 51309,
    "projectRank": 1,
    "givbackFactor": 0.75,
    "valueUsdAfterGivbackFactor": 0,
    "giverAddress": "0x7f37e3008207c27360b20abcfb5fdcc8e37596b8",
    "txHash": "0x7f37e3008207c27360b20abcfb5fdcc8e37596b8-0xe6375bc298aeb29d173b2ab359d492439a43b268-0xdfd824f6928b9776c031f7ead948090e2824ce8b-1.0-1.0-1727596303",
    "network": "optimism-sepolia",
    "source": "giveth.io",
    "giverName": "maryamaa mehr",
    "giverEmail": null,
    "projectLink": "https://giveth.io/project/mejan",
    "isReferrerGivbackEligible": false,
    "referrerWallet": null,
    "numberOfStreamedDonations": 21,
    "parentRecurringDonationId": "279",
    "parentRecurringDonationTxHash": "0x5d41c1bbc6268103fcd6becb6b99396f642a09cd65431fc46ff26e3ddbca40b4"
  },

**Comment by WhyldWanderer** [comment]
Hi team! @mohammadranjbarz & @maryjaf 

I can see that this was implemented for the eligible-donations API.. however I dont think that it is working correctly for the calculate-updated script. When I set the parameters to a min donation value of 4$ dollars, the recurring donations that total over 4$ are not included in the calculation.

**Comment by WhyldWanderer** [comment]
for instance there is this recurring donation that should be eligible, but it is not showing up in the calculation.

Giver Address: 0xd8d1a483637b31416d7d7f0c8cbb6a38695523e0
tx Hash: 0xd8d1a483637b31416d7d7f0c8cbb6a38695523e0-0x1d54f505b3df6db0e0458f6a1370c6541232f3ce-0x4cab5b9930210e2edc6a905b9c75d615872a1a7e-0.0-0.0-1725408108
parentRecurringDonationTxHash: 0x520d5fa309d2dd5ec7b327f2e006c198040b613964283f40fa758e5cccae9ec4

**Comment by mohammadranjbarz** [comment]
> for instance there is this recurring donation that should be eligible, but it is not showing up in the calculation.
> 
> Giver Address: 0xd8d1a483637b31416d7d7f0c8cbb6a38695523e0 tx Hash: 0xd8d1a483637b31416d7d7f0c8cbb6a38695523e0-0x1d54f505b3df6db0e0458f6a1370c6541232f3ce-0x4cab5b9930210e2edc6a905b9c75d615872a1a7e-0.0-0.0-1725408108 parentRecurringDonationTxHash: 0x520d5fa309d2dd5ec7b327f2e006c198040b613964283f40fa758e5cccae9ec4



@WhyldWanderer I think you have made a mistake there, you think those mini donations have `0.67$` but the `0.67` is givback factor and valudUsd is `0.045896333` there and sum of 14 of those donations would be `0.64` that is still less than `4$` so the script behave correctly. if you put `minEligibleValueUsd` to `0.5` you would see below item in the result

```
   {
        "giverAddress": "0xd8d1a483637b31416d7d7f0c8cbb6a38695523e0",
        "giverEmail": null,
        "giverName": "Makete Lagunak",
        "totalDonationsUsdValueAfterGivFactor": 0.2266976,
        "totalDonationsUsdValue": 0.6303113,
        "totalReferralDeductedUsdValue": 0,
        "totalReferralDeductedUsdValueAfterGivFactor": 0,
        "totalReferralAddedUsdValue": 0,
        "totalReferralAddedUsdValueAfterGivFactor": 0,
        "averageGivbackFactor": "0.3596597",
        "givback": 40.4194418,
        "givbackUsdValue": "0.2266976",
        "share": 0.0000124
      },
```

https://givbacks.giveth.io/calculate-updated?roundNumber=71&givAvailable=1000000&maxAddressesPerFunctionCall=200&nicePerDollar=1&niceWhitelistTokens=DAI%2CWXDAI%2CXDAI%2CUSDC%2CUSDC.e%2CUSDGLO&niceProjectSlugs=the-giveth-community-of-makers&minEligibleValueUsd=0.5&givethCommunityProjectSlug=the-giveth-community-of-makers&download=no

**Comment by WhyldWanderer** [comment]
Youre right @mohammadranjbarz! What a relief haha. Thanks for clarifying that for me 😓 I guess I got lost in the columns

**Comment by WhyldWanderer** [comment]
However now I am trying to prepare the spreadsheet for round 73 GIVbacks and it is still showing donations under 4$ even when I enter 4 in the min donation amount field. 

Here are the parameters that I used:
![image](https://github.com/user-attachments/assets/4f1f2af2-5011-450e-a08c-3b0a47e2b35e)

And here is the data that I got:
https://docs.google.com/spreadsheets/d/1fE6d3wAL_8dfA73ZdcFzd_nUCtv8Fv0inQqHRabqWh0/edit?usp=sharing

I tried doing it with both 4$ min and 0$ min so that I can compare the two. Youll find the two tabs at the bottom of the spreadsheet labeled accordingly.

Did I do something wrong in the API?
